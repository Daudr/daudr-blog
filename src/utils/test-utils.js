import React from "react"
import { configure, mount } from "enzyme"
import Adapter from "enzyme-adapter-react-16"

configure({ adapter: new Adapter() })

const TestHook = ({ callback }) => {
  callback()
  return null
}

export const testHook = callback => {
  mount(<TestHook callback={callback} />)
}
