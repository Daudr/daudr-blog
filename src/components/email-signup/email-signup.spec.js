import React from "react"
import renderer from "react-test-renderer"

import EmailSignup from "./email-signup"
import { testHook } from "../../utils/test-utils";

let invalidEmail

describe("EmailSignup", () => {
  beforeEach(() => {
    testHook(() => {
      
    })
  })

  it("renders correctly", () => {
    const tree = renderer
      .create(<EmailSignup />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
