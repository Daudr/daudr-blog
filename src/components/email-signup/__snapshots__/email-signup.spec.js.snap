// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmailSignup renders correctly 1`] = `
<div
  style={
    Object {
      "margin": "1.75rem",
      "textAlign": "center",
    }
  }
>
  <p
    style={
      Object {
        "marginBottom": 0,
      }
    }
  >
    Do you want to be updated when new articles are being published?
  </p>
  <p
    style={
      Object {
        "marginBottom": "0.875rem",
      }
    }
  >
    Join the newsletter!
     
    <span
      aria-label="sunglasses smile"
      role="img"
    >
      ðŸ˜Ž
    </span>
  </p>
  <form
    method="POST"
    style={
      Object {
        "alignItems": "baseline",
        "display": "flex",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="MuiFormControl-root"
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-marginNormal"
        data-cy="email-input"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="email"
        >
          Email
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            disabled={false}
            id="email"
            name="email"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="text"
            value=""
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline"
            style={
              Object {
                "paddingLeft": 8,
              }
            }
          >
            <legend
              className="PrivateNotchedOutline-legend-61"
              style={
                Object {
                  "width": 0.01,
                }
              }
            >
              <span
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "&#8203;",
                  }
                }
              />
            </legend>
          </fieldset>
        </div>
        
      </div>
      <p
        className="MuiFormHelperText-root"
        data-cy="error-text"
        hidden={true}
      >
        
      </p>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary Mui-disabled Mui-disabled"
      data-cy="email-button"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "height": "56px",
          "marginLeft": "0.875rem",
        }
      }
      tabIndex={-1}
      type="submit"
    >
      <span
        className="MuiButton-label"
      >
        Add me!
      </span>
    </button>
  </form>
</div>
`;
